Provide possibility to mask the __typename in client results (#533)
FormidableLabs/urql: .changeset/modern-queens-run.md

---
'@urql/core': minor
---

> Adds the `stripTypename` export to urql-core, this deeply strips typenames from the given payload. In (P)React we add an additional parameter to useQuery to strip typenames on result and strip typenames from variables in useRequest

Reviewer: Let’s get this updated :)

---
'@urql/core': minor
---

Adds the `maskTypename` export to urql-core, this deeply masks typenames from the given payload.

> Masking typenames has also become an option on the `client` under the name `maskTypename` setting this to `true` will automatically

Reviewer: ```suggestion
Masking `__typename` properties is also available as a `maskTypename` option on the `Client`. Setting this to true will
```

---
'@urql/core': minor
'@urql/preact': minor
'urql': minor
---

> Adds the `stripTypename` export to urql-core, this deeply makes `__typename` non-enumerable.

Reviewer: ```suggestion
Add the `stripTypename` export to `@urql/core` to make all `__typename` fields non-enumerable.
```

> In (P)React we'll apply the above function on result payloads of mutations, subscriptions and mutations.

Reviewer: ```suggestion
In `@urql/preact` and `urql` this is applied to all `data` results.
```
